src/Google/GoogleHelper.php

```diff
@@ -0,0 +1,1340 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Google;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\WC;
+
+/**
+ * Class GoogleHelper
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds
+ *
+ * @since x.x.x
+ */
+class GoogleHelper implements Service {
+
+	protected const SUPPORTED_COUNTRIES = [
+		// Algeria
+		'DZ' => [
+			'code'     => 'DZ',
+			'currency' => 'DZD',
+			'id'       => 2012,
+		],
+		// Angola
+		'AO' => [
+			'code'     => 'AO',
+			'currency' => 'AOA',
+			'id'       => 2024,
+		],
+		// Argentina
+		'AR' => [
+			'code'     => 'AR',
+			'currency' => 'ARS',
+			'id'       => 2032,
+		],
+		// Australia
+		'AU' => [
+			'code'         => 'AU',
+			'currency'     => 'AUD',
+			'id'           => 2036,
+			'subdivisions' => [
```

Comment:
Region codes can be identical to country codes, so it wouldn't be unique enough to put it all on one level. If it's to be done different I'd suggest to use the structure used by WC. One array for countries, and another for regions (indexed by country code).

Reply:
Thanks @mikkamp , I thought that regions codes that are compliant with [ISO 31662-2](https://en.wikipedia.org/wiki/ISO_3166-2) were unique between regions and countries.

When I checked Wikipedia I read this line: 
> Each complete ISO 3166-2 code can then be used to uniquely identify a country subdivision in a global context.

And that made me think that a global context was referring to regions and countries.

Reply:
Maybe the function comments need to be changed then. If we would actually be using ISO 3166-2 then for example Delaware would be [US-DE](https://en.wikipedia.org/wiki/ISO_3166-2:US) (which would be unique). But we are returning the code as `DE` which can be confused with Germany.

Reply:
Yeah, these are not actual ISO 3166-2 codes, and adding the comment was not a good idea. I've removed those since this is a mapping to WooCommerce state codes and nothing else.